src/Core/Product/ProductPresenter.php

```diff
@@ -796,4 +801,27 @@ protected function getProductEmbeddedAttributes(array $product)
 
         return $embeddedProductAttributes;
     }
+
+    /**
+     * Assemble the same features in one array
+     * @param $presentedProduct
+     * @return mixed
+     */
+    public function addFeaturesToDisplay($presentedProduct)
+    {
+        $features = array();
+        foreach ($presentedProduct['features'] as $feature) {
+            if (isset($feature['name'])) {
+                if (array_key_exists($feature['name'], $features)) {
+                    $features[$feature['name']]['value'] .= "\n" . $feature['value'];
+                } else {
+                    $features[$feature['name']] = $feature;
+                }
+            }
+        }
+
+        $presentedProduct['feature_to_display'] = $features;
```

Comment:
I'm not sure that "feature_to_display" is the best name for this property.

The way I see it, it's up to the view to decide how features are displayed, either grouped by feature name or otherwise. If for convenience, we decide to provide a property containing the features grouped by feature name, I believe it would be too opinionated to say that they are the "features to display".

In order to reflect that, I think we should rename "feature_to_display" to "grouped_features".